perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AttributesManagerBlImpl.java

```diff
@@ -1961,12 +1996,14 @@ public void setAttribute(PerunSession sess, Resource resource, Member member, At
 		if (!workWithUserAttributes) {
 			if (setAttributeWithoutCheck(sess, member, resource, attribute, false)) {
 				this.checkAttributeDependencies(sess, new RichAttribute<>(resource, member, attribute));
```

Comment:
This method has different order of methods inside than the others. It should be checks for syntax, semantics, dependencies.